### ISSUE TITLE
Fail to send notification by email with STARTTLS with PlainText

### ISSUE THREAD

**Issue by e9d26** [prometheus/alertmanager#1273]
**What I'd like to do**
send notification by email with STARTTLS with PlainText

**What I did**
send notification using openssl command and it makes **success!!** (executed command is below)
but, alertmanager failed to send notification with ***same credential information***.

```
printf "%s\0%s\0%s" <user> <pass> | openssl base64 -e | tr -d '\n'; echo  ## to get Auth plain text
openssl s_client -connect <host>:<port> -starttls smtp -ign_eof
AUTH PLAIN <auth plain text>
MAIL FROM: xxxx@yyyy.com
RCPT TO: xxxx@yyyy.com
DATA
SUBJECT 'TEST MESSAGE'
MESSAGE 'THIS IS A TEST'
.
QUIT
```

* Alertmanager Logs:
```
level=error ts=2018-03-05T08:48:52.555046949Z caller=api.go:803 msg="API error" err="bad_data: start time must be before end time; start time must be before end time; start time must be before end time; start time must be before end time"

level=error ts=2018-03-05T08:48:59.859725159Z caller=notify.go:303 component=dispatcher msg="Error on notify" err="starttls failed: x509: certificate is valid for xxxxx.com, not yyyyy.com"
```

* Alertmanager configuration file:
```
global:
 smtp_smarthost: '<host>'
 smtp_from: 'xxxx@yyyy.com'
 smtp_auth_username: '<user>'
 smtp_auth_password: '<pass>'
 smtp_require_tls: true

route:
  receiver: mail
  group_by: ['alertname','name','severity']
  group_wait: 30s 
  group_interval: 5m 
  repeat_interval: 5m
receivers:
- name: 'mail'
  email_configs:
    - send_resolved: true
      to: 'xxxx@yyyy.com'
```

Why did alertmanager fail to send notification???

**Comment by simonpasquier** [comment]
It seems that your server doesn't advertise a valid certificate:

```err="starttls failed: x509: certificate is valid for xxxxx.com, not yyyyy.com"```

Currently there's no way to bypass the verification so you'd need to fix the certificate.

There are a couple of PR opened to turn off the certificate verification or use a custom server certificate but all are stalled (see #1186, #1140 and #757).

**Comment by simonpasquier** [comment]
Closed by #1528
